src/wp-includes/fonts/class-wp-font-library.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Font Library class.
+ *
+ * This file contains the Font Library class definition.
+ *
+ * @package    WordPress
+ * @subpackage Fonts
+ * @since      6.4.0
+ */
+
+/**
+ * Font Library class.
+ *
+ * @since 6.4.0
+ */
+class WP_Font_Library {
+
+	/**
+	 * Font collections.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	private static $collections = array();
+
+	/**
+	 * Provides the expected mime-type value for font files per-PHP release.
+	 *
+	 * Due to differences in the values returned these values differ between PHP versions.
+	 *
+	 * This is necessary until a collection of valid mime-types per-file extension can be
+	 * provided to 'upload_mimes' filter.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return Array A collection of mime types keyed by file extension.
+	 */
+	public static function get_font_mime_types() {
+		$php_7_ttf_mime_type = PHP_VERSION_ID >= 70300 ? 'application/font-sfnt' : 'application/x-font-ttf';
+
+		return array(
+			'otf'   => 'application/vnd.ms-opentype',
+			'ttf'   => PHP_VERSION_ID >= 70400 ? 'font/sfnt' : $php_7_ttf_mime_type,
+			'woff'  => PHP_VERSION_ID >= 80100 ? 'font/woff' : 'application/font-woff',
+			'woff2' => PHP_VERSION_ID >= 80100 ? 'font/woff2' : 'application/font-woff2',
+		);
+	}
+
+	/**
+	 * Registers a new font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $config Font collection config options.
+	 *                      See {@see wp_register_font_collection()} for the supported fields.
+	 * @return WP_Font_Collection|WP_Error A font collection is it was registered successfully and a WP_Error otherwise.
+	 */
+	public static function register_font_collection( $config ) {
+		$new_collection = new WP_Font_Collection( $config );
+
+		if ( isset( self::$collections[ $config['id'] ] ) ) {
+			return new WP_Error( 'font_collection_registration_error', __( 'Font collection already registered.' ) );
+		}
+
+		self::$collections[ $config['id'] ] = $new_collection;
+
+		return $new_collection;
+	}
+
+	/**
+	 * Gets all the font collections available.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return array List of font collections.
+	 */
+	public static function get_font_collections() {
+		return self::$collections;
+	}
+
+	/**
+	 * Gets a font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $id Font collection id.
+	 * @return array List of font collections.
+	 */
+	public static function get_font_collection( $id ) {
+		if ( array_key_exists( $id, self::$collections ) ) {
+			return self::$collections[ $id ];
+		}
+
+		return new WP_Error( 'font_collection_not_found', 'Font collection not found.' );
```

Comment:
String needs to be translated

Reply:
Addressing in https://github.com/WordPress/gutenberg/pull/58675